shared/src/commonMain/kotlin/com/example/tasky/dataSource/AgendaDataSource.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.tasky.dataSource
+
+import com.example.tasky.manager.HttpManager
+import com.example.tasky.model.BaseError
+import com.example.tasky.model.DataError
+import com.example.tasky.model.ErrorResponse
+import com.example.tasky.model.ResultWrapper
+import com.example.tasky.model.agenda.Agenda
+import com.example.tasky.model.agenda.GetAgendaResponse
+import com.example.tasky.util.isSuccess
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.plugins.resources.get
+import io.ktor.client.request.bearerAuth
+import io.ktor.client.request.parameter
+import kotlin.coroutines.cancellation.CancellationException
+
+class AgendaDataSource(
+    private val httpClient: HttpClient = HttpManager.httpClient,
+) {
+    suspend fun getAgenda(
+        timeStamp: Long,
+        token: String,
+    ): ResultWrapper<GetAgendaResponse, BaseError> {
+        return try {
+            val response =
+                httpClient.get(Agenda()) {
+                    bearerAuth(token)
+                    parameter("time", timeStamp)
+                }
+
+            if (!response.isSuccess()) {
+                val errorResponse = response.body<ErrorResponse>()
+                println("get agenda fail: ${errorResponse.message}")
+                return ResultWrapper.Error(DataError.Remote.UNKNOWN)
```

Comment:
Could it make more sense to map the response code to a specific error instead of returning `DataError.Remote.UNKNOWN`? And since you'll have a lot of these API calls that can all give you similar errors, could it make sense to extract this mapping into a function? Something like this, maybe?


```
suspend inline fun <reified T> responseToResult(response: HttpResponse): Result<T, DataError.Network> {
    return when (response.status.value) {
        in 200..299 -> Result.Success(response.body<T>())
        401 -> Result.Error(DataError.Network.UNAUTHORIZED)
        408 -> Result.Error(DataError.Network.REQUEST_TIMEOUT)
        409 -> Result.Error(DataError.Network.CONFLICT)
        413 -> Result.Error(DataError.Network.PAYLOAD_TOO_LARGE)
        429 -> Result.Error(DataError.Network.TOO_MANY_REQUESTS)
        in 500..599 -> Result.Error(DataError.Network.SERVER_ERROR)
        else -> Result.Error(DataError.Network.UNKNOWN)
    }
}
```